application/client/src/app/ui/views/sidebar/search/providers/providers.ts

```diff
@@ -497,9 +502,90 @@ export class Providers {
                 event.items = event.items.concat(custom);
             }
         });
+
+        this.contextMenuOptions(event.items);
+
         this.subjects.get().context.emit(event);
     }
 
+
+    public contextMenuOptions(items: IMenuItem[]): IMenuItem[] {
+        const historySession = history.get(this.session);
+        if(historySession === undefined) {
+            this.logger.error('History session is not defined');
+            return items;
+        }
+        items.push({ /* Delimiter */ });
+        const store = this.session.search.store();
+        const showExport: boolean = (store.filters().get().length + store.charts().get().length + store.disabled().get().length) !== 0;
+        showExport && items.push(
+        {
+            caption: 'Export All to File',
+            handler: () => this.filters(historySession).export(),
+        });
+        items.push(
+        {
+            caption: 'Import from File',
+            handler: () => this.filters(historySession).import(),
+        });
+        return items;
+    }
+
+    protected filters(historySession: HistorySession): { import(): void; export(): void } {
+        return {
```

Comment:
method `logAndNotifyError` can be present as arrow-function here (as soon as it's used only in this scope)
```
const logAndNotifyError = () => {...};
return {...}
```